algosdk/dryrun_results.py

```diff
@@ -0,0 +1,85 @@
+from typing import List
+
+
+class DryrunResponse:
+    def __init__(self, drrjson: dict):
+        # These are all required fields
+        self.error = drrjson["error"]
+        self.protocol = drrjson["protocol-version"]
+        self.txns = [DryrunTransactionResult(txn) for txn in drrjson["txns"]]
+
+
+class DryrunTransactionResult:
+    def __init__(self, dr):
+        # Only required field
+        self.disassembly = dr["disassembly"]
+
+        # These are all optional
+        if "app-call-messages" in dr:
+            self.app_call_messages = dr["app-call-messages"]
+        if "app-call-trace" in dr:
+            self.app_call_trace = DryrunTrace(dr["app-call-trace"])
+        if "local-deltas" in dr:
+            self.local_deltas = dr["local-deltas"]
+        if "global-delta" in dr:
+            self.global_delta = dr["global-delta"]
+        if "cost" in dr:
+            self.app_call_cost = dr["cost"]
+        if "logic-sig-messages" in dr:
+            self.logic_sig_messages = dr["logic-sig-messages"]
+        if "logic-sig-trace" in dr:
+            self.logic_sig_trace = DryrunTrace(dr["logic-sig-trace"])
+        if "logs" in dr:
+            self.logs = dr["logs"]
+
+    def app_trace(self, trace_spaces: int = 16):
```

Comment:
Is there a reason for the default of 16? In any case, I recommend making this "magic number" a class constant.
Additionally, in the spirit of D.R.Y. (don't repeat yourself) I recommend adding a new class method which does the string formatting and is called by each of the `*_trace()` instance methods:
Something like:
```python
class DryRunTransactionResult:
    DEFAULT_TRACE_SPACES = 16
... etc ...
    @classmethod
    trace(cls, dr_trace: DryRunTrace, spaces: int = None):
        if spaces is None:
            spaces = cls.DEFAULT_TRACE_SPACES
        for line in dr_trace.get_trace():
            ... etc ...

 
        def app_trace(self, spaces: int = None):
            return self.trace(self.app_call_trace, spaces=spaces)

        .... similarly for lsig_trace() ...